### PR TITLE
refactored research-hub templates into a namespaced directory

### PR DESCRIPTION
**Pull request by danielfmiranda**
# Description

Related PRs/issues: #10393

As requested in #10393, this PR does the following:
- Removes the prepended `research-` from all of the templates that were related to the research hub
- Moves these templates to a more central directory of `network-api/networkapi/templates`, within a subdirectory of `research-hub/` 

**Note:** The template of `breadcrumbs.html` has been moved to `network-api/networkapi/templates` but **not** in a subdirectory of `research-hub`. This is because the template `blog_author_detail_page.html` which is not research-hub related also uses the template.

### CODE REVIEW

File: network-api/networkapi/templates/fragments/breadcrumbs.html
Comment (jhonatan-lopes) [inline network-api/networkapi/templates/fragments/breadcrumbs.html:1]: Is this `breadcrumbs` fragments applicable to all pages, even outside the `research_hub`?

File: network-api/networkapi/wagtailpages/pagemodels/research_hub/library_page.py [hunk 1]
```diff
@@ -57,7 +62,7 @@ class ResearchLibraryPage(research_base.ResearchHubBasePage):
         SynchronizedField("search_image"),
     ]
 
-    def get_context(self, request: "http.HttpRequest") -> "template.Context":
+    def get_context(self, request: "http.HttpRequest") -> "django_template.Context":
```
Comment (jhonatan-lopes) [inline network-api/networkapi/wagtailpages/pagemodels/research_hub/library_page.py:65]: Was this an import conflict? If so, it must have been hard to track down :face_exhaling:

**Review by jhonatan-lopes — CHANGES_REQUESTED** [review]
Thanks for this @danielfmiranda, it all looks good. The only issue I have is with the folder name being `research-hub` which is not consistent with the ones that are on `pagemodels` already, which are using `research_hub`.

For consistency, can we change `research-hub` to `research_hub`, please?

File: network-api/networkapi/wagtailpages/pagemodels/research_hub/library_page.py (hunk 1 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/research_hub/library_page.py:65]: Hi @jhonatan-lopes, yes it was! However, after importing `template` as `django_template` the error was resolved.

File: network-api/networkapi/templates/fragments/breadcrumbs.html
Comment (danielfmiranda) [reply network-api/networkapi/templates/fragments/breadcrumbs.html:1]: Hi @jhonatan-lopes, yes it is! It is also being used by another template `blog_author_detail_page.html`. 

For that reason, I have decided to move it, but without nesting it inside a  `research-hub/` subdirectory, as it is not directly related to it 👍

**Comment by danielfmiranda** [comment]
Hi @jhonatan-lopes thanks for the review and feedback! 

Regarding your request to update the directory name from `research-hub` to `research_hub`, I would actually vote to leave the directory name as-is, as it appears that the naming convention for html templates seems to be: `directory/sub-directory/file_name.html`.  

Maybe @mmmavis can chime in if this is in fact the case?


Here is a screenshot of the file tree showing what I mean, where the subdirectories use `dashes-between` and the file names `use_underscores_between`:

Screenshot 2023-04-24 at 11 00 03 AM

**Comment by danielfmiranda** [comment]
Hi @jhonatan-lopes, per the conversation in this [slack thread](https://mozilla.slack.com/archives/C043Q45BBMF/p1682362781329819,) I have updated the template directory names from `research-hub` to `research_hub` like requested.

I will create a ticket to update the other directories to use underscores as well, thanks! 👍

**Comment by danielfmiranda** [comment]
Hi @jhonatan-lopes, per the conversation in this [slack thread](https://mozilla.slack.com/archives/C043Q45BBMF/p1682362781329819,) I have updated the template directory names from `research-hub` to `research_hub` like requested.

I will create a ticket to update the other directories to use underscores as well.

PS: Please excuse the messy timeline above, I had to do some troubleshooting as tests were failing, but I found the typo and everything is ready for re-review. Thanks! 👍

File: network-api/networkapi/templates/fragments/breadcrumbs.html
Comment (jhonatan-lopes) [reply network-api/networkapi/templates/fragments/breadcrumbs.html:1]: That's great, less work for us to do on #10420. Thanks @danielfmiranda!

**Comment by jhonatan-lopes** [comment]
Thanks for the updates @danielfmiranda! It all looks great!

**Review by jhonatan-lopes — APPROVED** [review]

**Pull request merged**